pkg/operators/ebpf/formatters.go

```diff
@@ -92,18 +94,42 @@ func (i *ebpfInstance) initEnumFormatter(gadgetCtx operators.GadgetContext) erro
 				return err
 			}
 
-			formatter := func(ds datasource.DataSource, data datasource.Data) error {
-				// TODO: lookup table?
-				inBytes := in.Get(data)
-				val := byteSliceAsUint64(inBytes, enum.Signed, ds)
-				for _, v := range enum.Values {
-					if val == v.Value {
-						out.Set(data, []byte(v.Name))
-						return nil
+			var formatter func(ds datasource.DataSource, data datasource.Data) error
+
+			isBitField := strings.HasSuffix(enum.Name, "_set")
+			if isBitField {
+				separator := in.Annotations()[enumBitfieldSeparatorAnnotation]
+				if separator == "" {
+					separator = "|"
+				}
+
+				formatter = func(ds datasource.DataSource, data datasource.Data) error {
+					inBytes := in.Get(data)
+					val := byteSliceAsUint64(inBytes, enum.Signed, ds)
+
+					var arr []string
+					for _, v := range enum.Values {
+						if val&v.Value != 0 {
```

Comment:
```suggestion
						if val&v.Value == v.Value {
```

This change is important for combinations. If for example you've got an enum entry with a value of 0x03, it otherwise would be `!= 0` for 0x01 and 0x02.

Reply:
I was wondering about this since there was a similar comment in https://github.com/inspektor-gadget/inspektor-gadget/pull/3025#discussion_r1648749330. Thanks for catching this up.